net/Cargo.toml

```diff
@@ -27,6 +27,8 @@ snowflake = "=1.3.0"
 tungstenite = "=0.6.1"
 url = "=1.7.2"
 native-tls = "=0.2.2"
+crossbeam-channel = "=0.3.8"
+parking_lot = "=0.8.0"
```

Comment:
Both are consequences of making `NetHandler` `Send` and `Sync`.

`crossbeam-channel` is replacing std channels mostly in tests where the sender is passed into the `NetHandler` which can't work with std `Senders` because they are not `Send`.

`parking_lot` is needed since the `NetHandler` is used recursively in the same thread which was fine before, but if we wrap it with an std `RwLock` it can't detect that it is the same thread and will block which creates deadlocks.

What are downsides to using these crates for these purposes?

Reply:
I don't have the rust expertise to suggest whether it even has a downside, or even if there is a good alternative to crossbeam or not. Just wanted to make sure the dependency introduction is considered in full. @dymayday any opinions on this?

Reply:
BTW `parking_lot` is about to replace the sync implementations of the std lib.
Discussion: https://internals.rust-lang.org/t/standard-library-synchronization-primitives-and-undefined-behavior/8439/22
PR for the replacement: https://github.com/rust-lang/rust/pull/56410

I would actually propose that we default to `parking_lot` whenever we need sync primitives. Those are faster than std and reentrant.

Reply:
And there is recent discussion to replace `std::mpsc` with `crossbeam-channel`: https://stjepang.github.io/2019/03/02/new-channels.html